sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -160,6 +160,9 @@ class OrcFileFormat
     val capacity = sqlConf.orcVectorizedReaderBatchSize
 
     OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE.setBoolean(hadoopConf, sqlConf.caseSensitiveAnalysis)
+    OrcConf.FORCE_POSITIONAL_EVOLUTION.setBoolean(hadoopConf,
+      hadoopConf.getBoolean("spark.hadoop." + OrcConf.FORCE_POSITIONAL_EVOLUTION.getAttribute,
+        false))
```

Comment:
This is added at the last commit. However, we already remove `spark.hadoop.` prefix from Spark side already, don't we?
- https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala#L96-L106

Reply:
Please see line 162. `hadoopConf` is a cleaned one.

Reply:
Oh I see. Yeah, you're right. The hadoop conf here is cleaned up. If users specify `spark.hadoop.orc.force.positional.evolution`, it will be removed the prefix and added into this conf.

Reply:
Hmm, I might get something wrong but I don't think that `hadoopConf` is cleaned here as if I remove this line the UTs fail. As far as I see the only reference to `SparkHadoopUtil.appendSparkHadoopConfigs` is from`spark-hive` [HiveUtils.newTemporaryConfiguration](https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala#L547).

I can revert the prefix in https://github.com/apache/spark/pull/29737/files#diff-e14fd8725cf71eee7b34fa299c2f3abe5a0033f9abce9de4c7e081ba57991b0bR146 though.

Reply:
@peter-toth . You should not write a UT like `withSQLConf("spark.hadoop.xxx")`. `withSQLConf` is a testing framework only.

Reply:
Parquet is the same. Please see `ParquetFileFormat.scala`. In this context, the Hadoop Conf should have the original conf key, not `spark.hadoop` prefixed ones.
```
      lazy val footerFileMetaData =
        ParquetFileReader.readFooter(sharedConf, filePath, SKIP_ROW_GROUPS).getFileMetaData
```

Reply:
Thanks @dongjoon-hyun, I see it now where the cleaning happens and reverted the last commit.